mldsa/poly.c

```diff
@@ -320,6 +327,19 @@ uint32_t mld_poly_chknorm(const mld_poly *a, int32_t B)
     invariant((t == 0) == array_abs_bound(a->coeffs, 0, i, B))
   )
   {
+    /*
+     * Since we know that -REDUCE32_RANGE_MAX <= a < REDUCE32_RANGE_MAX,
+     * and B <= MLDSA_Q - REDUCE32_RANGE_MAX, to check if
+     * -B < (a mod± MLDSA_Q) < B, it suffices to check if -B < a < B.
+     * In other words: If -B < a < B, then for k >= 1, we also have
+     * -B < (a ± k*MLDSA_Q) < B.
+     *
+     * We prove this to be true using the following CBMC assertions.
+     * a ==> b expressed as !a || b to also allow run-time assertion.
+     */
```

Comment:
This comment should be moved into the @Ref section as it's something one needs to know when validating that the spec in the code is equivalent to the spec in FIPS-204.

I would also emphasize the main point:
```
The definition of this FIPS-204 requires signed canonical reduction prior 
to applying the bounds check. However, `-B < (a mod± MLDSA_Q) < B`
is equivalent to `-B < a < B` under the assumption that 
`B <= MLDSA_Q - REDUCE32_RANGE_MAX` (cf. the assertion in below).
Hence, the present spec and implementation are correct without reduction.
```

Reply:
Okay. I moved it to the header and adjusted it as suggested.